packages/starknet-snap/src/executeTxn.ts

```diff
@@ -1,28 +1,73 @@
 import { toJson } from './utils/serializer';
-import { getNetworkFromChainId, getTxnSnapTxt } from './utils/snapUtils';
-import { getKeysFromAddress, executeTxn as executeTxnUtil } from './utils/starknetUtils';
-import { ApiParams, ExecuteTxnRequestParams } from './types/snapApi';
+import { CallDetails } from 'starknet';
+import { getNetworkFromChainId, getTxnSnapTxt, addDialogTxt } from './utils/snapUtils';
+import { getKeysFromAddress, executeTxn as executeTxnUtil, isAccountDeployed } from './utils/starknetUtils';
+import { ApiParams, EstimateFeeRequestParams, ExecuteTxnRequestParams } from './types/snapApi';
+import { createAccount } from './createAccount';
 import { DialogType } from '@metamask/rpc-methods';
-import { heading, panel } from '@metamask/snaps-sdk';
+import { heading, panel, divider } from '@metamask/snaps-sdk';
 import { logger } from './utils/logger';
+import { estimateFee } from './estimateFee';
 
 export async function executeTxn(params: ApiParams) {
   try {
     const { state, keyDeriver, requestParams, wallet } = params;
     const requestParamsObj = requestParams as ExecuteTxnRequestParams;
+    const requestParamsObjEstimateFee = requestParams as EstimateFeeRequestParams;
 
-    logger.log(`executeTxn params: ${toJson(requestParamsObj, 2)}}`);
+    const txnInvocation = (
```

Comment:
it looks like we can only support 1 txn

Reply:
The code should now work correctly with multiple transactions